### PR TITLE
add readme

### PR DESCRIPTION
**Pull request by ktaebum**

### CODE REVIEW

**Review by Algy — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,11 +1,235 @@
...
+#### Contents
+
+1. [Installation](#installation)
+2. [Using PeriFlow SDK](#using-periflow-sdk)
+3. [Examples](#examples)
+
+### Installation
+
+PeriFlow SDK is built to PyPI and can be installed as follows:
+```
```
Comment (kooyunmo) [inline README.md:19]:
```sh

Comment (kooyunmo) [inline README.md:19]:
```suggestion
```sh
```

File: README.md [hunk 2]
```diff
@@ -1,11 +1,235 @@
...
+pip install periflow_sdk
+```
+Also, you can install from source by cloning this repository:
 ```
```
Comment (kooyunmo) [inline README.md:23]:
```suggestion
```sh
```

File: README.md [hunk 3]
```diff
@@ -1,11 +1,235 @@
...
-python setup.py install
-python
-> import periflow_sdk
+pip install .
+```
+
+### Using PeriFlow SDK
+
+Basically, you can import and use our SDK as follows:
+```
```
Comment (kooyunmo) [inline README.md:32]:
```suggestion
```python
```

File: README.md [hunk 4]
```diff
@@ -1,11 +1,235 @@
...
+import periflow_sdk as pf
+```
+
+PeriFlow SDK provides several functions to make your training code works with PeriFlow.
+
+```
+1. pf.init(total_train_steps: int, local_log_name: Optional[str] = None) -> None
+```
```
Comment (kooyunmo) [inline README.md:40]:
```suggestion
```python
pf.init(total_train_steps: int, local_log_name: Optional[str] = None) -> None
```
```

File: README.md [hunk 5]
```diff
@@ -1,11 +1,235 @@
...
+- `total_train_steps`: The number of total training steps of the training job
+- `local_log_name`: local filename where `pf.metric` writes into
+
+**Note**: `local_log_name` is meaningful to local mode
+
+---
+
+```
+2. pf.start_step() -> None
+```
```
Comment (kooyunmo) [inline README.md:54]:
```suggestion
```python
pf.start_step() -> None
```
```

File: README.md [hunk 6]
```diff
@@ -1,11 +1,235 @@
...
+Mark that a single training step begins.
+
+---
+
+```
+3. pf.end_step() -> None
+```
```
Comment (kooyunmo) [inline README.md:61]:
```suggestion
```python
pf.end_step() -> None
```
```

File: README.md [hunk 7]
```diff
@@ -1,11 +1,235 @@
...
+Mark that the single training step ends.
+
+**Note**: we provide the function `pf.train_step`, a contextmanager which wraps `start_step` and `end_step`. `train_step` can be used as
+```
+with pf.train_step():
+    # your training step code
+```
```
Comment (kooyunmo) [inline README.md:68]:
```suggestion
```python
with pf.train_step():
    # your training step code
```
```

File: README.md [hunk 8]
```diff
@@ -1,11 +1,235 @@
...
+
+---
+
+```
+4. pf.upload_checkpoint() -> None
+```
```
Comment (kooyunmo) [inline README.md:74]:
```suggestion
```python
4. pf.upload_checkpoint() -> None
```
```

File: README.md [hunk 9]
```diff
@@ -1,11 +1,235 @@
...
+**Note**: In distributed training (e.g., torch DDP), all ranks should call this function even if some of them have not actually saved the checkpoint.
+For example, not
+```
+import torch
+
+...
+if torch.distributed.get_rank() == 0:
+    torch.save(state_dict, CKPT_PATH)
+    pf.upload_checkpoint()
+```
```
Comment (kooyunmo) [inline README.md:87]:
```suggestion
```python
import torch

...
if torch.distributed.get_rank() == 0:
    torch.save(state_dict, CKPT_PATH)
    pf.upload_checkpoint()
```
```

File: README.md [hunk 10]
```diff
@@ -1,11 +1,235 @@
...
+, but
+```
```
Comment (kooyunmo) [inline README.md:89]:
```suggestion
```python
```

File: README.md [hunk 11]
```diff
@@ -1,11 +1,235 @@
...
+if torch.distributed.get_rank() == 0:
+    torch.save(state_dict, CKPT_PATH)
+pf.upload_checkpoint()
+```
+
+---
+
+```
+5. pf.metric(msg: Dict[str, JSONValue]) -> None
+```
```
Comment (kooyunmo) [inline README.md:102]:
```suggestion
```python
pf.metric(msg: Dict[str, JSONValue]) -> None
```
```

**Review by kooyunmo — CHANGES_REQUESTED** [review]
Please add a language specification for the code blocks

**Comment by kooyunmo** [comment]
For me, the code block located inside a sentence looks weird. Can you please complete the sentences and put the example code block after each sentence?

**Review by Algy — APPROVED** [review]

**Review by kooyunmo — APPROVED** [review]

**Pull request merged**